cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,507 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+Copyright 2015 CoreOS, Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/client/leaderelection"
+	"k8s.io/kubernetes/pkg/client/record"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/flag"
+	nodeutil "k8s.io/kubernetes/pkg/util/node"
+	"k8s.io/kubernetes/pkg/util/tpm"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	TaintKey       string = "Untrusted"
+	LogState       string = "tpm.coreos.com/logstate"
+	UntrustedSince string = "tpm.coreos.com/untrustedsince"
+	TrustedSince   string = "tpm.coreos.com/trustedsince"
+	ValidationTime string = "tpm.coreos.com/validationtime"
+	ConfigName     string = "tpm-manager.coreos.com"
+)
+
+// Flag a node as untrusted
+func invalidateNode(node *api.Node) error {
+	if node.ObjectMeta.Annotations == nil {
+		node.ObjectMeta.Annotations = make(map[string]string)
+	}
+	newTaints := []api.Taint{}
+	untrustedTaint := api.Taint{
+		Key:    TaintKey,
+		Value:  "true",
+		Effect: api.TaintEffectNoSchedule,
```

Comment:
I don't believe you can move an existing pod explicitly. It would be out of the normal operations if you could, but I'd have to double-check to make sure.